frontend/src/app/components/comparison/comparison.component.ts

```diff
@@ -0,0 +1,183 @@
+import { StockPriceService } from 'src/app/services/stock-price.service';
+import * as Highcharts from 'highcharts';
+
+@Component({
+  selector: 'app-comparison',
+  templateUrl: './comparison.component.html',
+  styleUrls: ['./comparison.component.css']
+})
+export class ComparisonComponent implements OnInit {
+
+  public state:string;
+  public stockPrices:StockPrice[][];
+  public companies:Company[];
+  public exchanges:Exchange[];
+  public companyTitle:string;
+  public exchangeTitle:string;
+  public companySelected:Company;
+  public exchangeSelected:Exchange;
+  public fromTime:string;
+  public toTime:string;
+
+  highcharts = Highcharts;
+  chartOptions: Highcharts.Options;
+
+  constructor(private authService:AuthService, private companyService:CompanyService, private exchangeService:ExchangeService, private stockPriceService:StockPriceService) {
+    this.state="";
+    this.stockPrices = [];
+    this.companyTitle="Please choose a company";
+    this.exchangeTitle="Please choose a stock exchange";
+    this.companies=[];
+    this.exchanges=[];
+    this.fromTime="";
+    this.toTime="";
+    this.companySelected = {
+      "id": 0,
+      "name": "",
+      "turnover": 0,
+      "ceo": "",
+      "brief": "",
+      "bod": "",
+      "sector": {
+          "id": 0,
+          "name": "",
+          "brief": ""
+      }
+    }
+    this.exchangeSelected = {
+      "id": 0,
+      "name": "",
+      "brief": "",
+      "remarks": "",
+      "address": {
+          "id": 0,
+          "street": "",
+          "city": "",
+          "country": "",
+          "zipCode": 0
+      }
+    }
+
+    this.chartOptions = {
+      title: {
+        text: "Stock Value Comparison"
+      },
+      xAxis: {
+        labels: {
+          formatter: function () {
+            var label = this.axis.defaultLabelFormatter.call(this).replace(" ","").replace("k","000").replace(" ","").replace(" ","").replace("M", "000000");
+            var date: Date = new Date(parseInt(label));
+            return date.getDate()+"/"+date.getMonth()+"/"+date.getFullYear()+"\n"+date.getHours()+":"+date.getMinutes()+":"+date.getSeconds();
+          }
+        }
+      },
+      yAxis: {
+        title: {
+          text: "Stock Price"
+        }
+      },
+      series: [],
+      legend: {
+        title: {
+            text: 'Stock<br/><span style="font-size: 9px; color: #666; font-weight: normal">(Click to hide)</span>',
+            style: {
+                fontStyle: 'italic'
+            }
+        },
+        layout: 'vertical',
+        align: 'right',
+        verticalAlign: 'top',
+        x: -10,
+        y: 100
+      }
+    }
+
+  }
+
+  ngOnInit(): void {
+    this.state = this.authService.getCurrentUserRole();
+    this.companyService.getAllCompanies().subscribe(companies => {
+      this.companies = companies;
+    });
+    this.exchangeService.getAllExchanges().subscribe(exchanges => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/ibmpariwesh/socgen-stockmarket-project/pullRequest?prid=8405405)
```suggestion
    this.exchangeService.getAllExchanges().subscribe((exchanges) => {
```